Source/dx.cpp

```diff
@@ -241,7 +241,7 @@ void lock_buf_priv()
 	if (error_code != DD_OK)
 		DDErrMsg(error_code, 235, "C:\\Src\\Diablo\\Source\\dx.cpp");
 
-	gpBufEnd += (int)ddsd.lpSurface;
+	gpBufEnd += (uintptr_t)ddsd.lpSurface;

```

Comment:
My version of this cleanup uses a `DWORD` here. uintptr_t doesn't work with VC6.

Reply:
Dang, simply casting a memory address to an int doesn't work for 64bit builds, I guess we will just carry the change in devilutionx.

Reply:
I removed `uintptr_t` from the pr.

Reply:
Here's what the header file looks like. Perhaps you could do a define for x64 and do `unsigned __int64` instead?
```
#ifdef _WIN64
 typedef unsigned __int64 __uintptr_t;
#else
 typedef unsigned int __uintptr_t;
#endif
typedef __uintptr_t uintptr_t;
```

Reply:
For Devilution (not DevilutionX)?, sure.

Reply:
Cast to size_t instead.

Reply:
Sounds like a good solution :+1: